cap/err_tolerance.go

```diff
@@ -0,0 +1,56 @@
+package cap
+
+import (
+	"time"
+)
+
+// ErrToleranceResult indicates the result of a error tolerance check
+type ErrToleranceResult uint32
+
+const (
+	// ErrToleranceSurpassed indicates the error tolerance has been surpassed
+	ErrToleranceSurpassed = iota
+	// IncreaseErrCount indicates that we should allow the error to happen
```

Comment:
Why not call it `AllowError`? Do you mean something like 

`IncreaseErrorCount will increase the count of errors observed`?